From cc9d1ed61c2ce46c5e4df336be74b89d41fab260 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tqs.de>
Date: Wed, 28 May 2014 14:41:49 +0200
Subject: [PATCH] ARM: tqma28: add machine init handler

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tqs.de>
Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 arch/arm/mach-mxs/mach-mxs.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/arch/arm/mach-mxs/mach-mxs.c
+++ b/arch/arm/mach-mxs/mach-mxs.c
@@ -341,6 +341,12 @@ static void __init m28cu3_init(void)
 	update_fec_mac_prop(OUI_DENX);
 }
 
+static void __init tqma28_init(void)
+{
+	enable_clk_enet_out();
+	/* TODO: update_fec_mac_prop(OUI_TQS); */
+}
+
 static const char __init *mxs_get_soc_id(void)
 {
 	struct device_node *np;
@@ -477,6 +483,8 @@ static void __init mxs_machine_init(void
 		m28cu3_init();
 	else if (of_machine_is_compatible("i2se,duckbill"))
 		duckbill_init();
+	else if (of_machine_is_compatible("tqs,tqma28"))
+		tqma28_init();
 
 	of_platform_populate(NULL, of_default_bus_match_table,
 			     NULL, parent);
