From e819ea49ca3f2cef8696a5cf408a49fab14deb02 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tqs.de>
Date: Wed, 28 May 2014 14:42:09 +0200
Subject: [PATCH] tqma28x: add device trees to makefile

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tqs.de>
---
 arch/arm/boot/dts/Makefile |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 0320303..1ffa476 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -186,6 +186,9 @@ dtb-$(CONFIG_ARCH_MXS) += imx23-evk.dtb \
 	imx28-evk.dtb \
 	imx28-m28cu3.dtb \
 	imx28-m28evk.dtb \
+	imx28-mba28.dtb \
+	imx28-mba28l-aa.dtb \
+	imx28-mba28l-ab.dtb \
 	imx28-sps1.dtb \
 	imx28-tx28.dtb
 dtb-$(CONFIG_ARCH_NOMADIK) += ste-nomadik-s8815.dtb
-- 
1.7.10.4

